components-rs/log.rs

```diff
@@ -0,0 +1,118 @@
+use std::cell::RefCell;
+use std::collections::BTreeSet;
+use std::ffi::c_char;
+use std::slice;
+use bitflags::bitflags;
+use ddcommon_ffi::CharSlice;
+use ddcommon_ffi::slice::AsBytes;
+
+bitflags! {
+    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+    #[repr(C)]
+    pub struct Log: u32 {
+        const None = 0;
```

Comment:
you reinvent log levels :)

Reply:
Yeah, though I need log levels interoperable between Rust and C - assuming we have rust code which will need logging.

Reply:
I still think we should have `None`, `Debug`, `Info`, `Warn` and `Error` I believe. I think once can be a nice addition indeed. 
`Deprecated` to me would be a `once` `Warning` and `Startup` is orthogonal I believe so should be removed

Reply:
I agree that `deprecated` is a `once` `warning` level. At the bitfield level I'm okay with Startup being smushed in there, but if we start using this system, I'd like to see a bit more separation in these concepts at the API level.